### PR TITLE
CONSOLE-4292: Console-operator should configure console with the CSP allowed directives

### PR DESCRIPTION
**Pull request by jhadvig**
This PR is vendoring API change for [adding new API field to ConsolePlugin CRD for allowing additional CSP sources](https://github.com/openshift/enhancements/pull/1706) and aggregating the CSP directives into console-config.yaml which will be used for console backend configuration.
The overall change is being described[ this enhancement doc change](https://github.com/openshift/enhancements/pull/1706)

Changes made:
- [x] vendor API changes
- [x] fix library-go and client-go API changes
- [x] add logic for CSP aggregation
- [x] add unit tests
- [x]  add e2e tests

/assign @TheRealJon @vojtechszocs

### CODE REVIEW

File: pkg/console/subresource/consoleserver/config_builder.go [hunk 1]
```diff
@@ -244,19 +251,20 @@ func (b *ConsoleServerCLIConfigBuilder) CopiedCSVsDisabled(copiedCSVsDisabled bo
...
+		Session:               b.session(),
+		ClusterInfo:           b.clusterInfo(),
+		Customization:         b.customization(),
+		ServingInfo:           b.servingInfo(),
+		Providers:             b.providers(),
+		MonitoringInfo:        b.monitoringInfo(),
+		Plugins:               b.plugins(),
+		I18nNamespaces:        b.i18nNamespaces(),
+		Proxy:                 b.proxy(),
+		ContentSecurityPolicy: b.contentSecurityPolicy(),
```
Comment (jhadvig) [inline pkg/console/subresource/consoleserver/config_builder.go:266]: @TheRealJon eventually we should unify the plugins related field (`Plugins`, `I18Namespaces`, `Proxy` and `ContentSecurityPolicy`) under a single field eg. `PluginInfo`, for consistency.

**Review by TheRealJon — APPROVED** [review]
Just one missed comment, otherwise:

/lgtm

File: pkg/console/operator/operator.go [hunk 2]
```diff
@@ -87,6 +87,7 @@ type consoleOperator struct {
 	versionGetter             status.VersionGetter
 	// lister
 	consolePluginLister listerv1.ConsolePluginLister
+	// feature gate
```
Comment (TheRealJon) [inline pkg/console/operator/operator.go:90]: nit, what is this referring to?
```suggestion
```

**Review by TheRealJon — APPROVED** [review]
/lgtm

**Comment by jhadvig** [comment]
flake 
/retest

**Comment by yapei** [comment]
tested together with [openshift/console#14481](https://github.com/openshift/console/pull/14481), no issues found

**Pull request merged**